### PR TITLE
Enhancements to prepare generic job scheduling

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -442,9 +442,18 @@ const setUpApi = (webSocketServer, xo, verboseLogsOnErrors) => {
   })
 }
 
+const setUpJobExecutor = xo => {
+  const executor = new JobExecutor(xo)
+  xo.executor = executor
```
Comment (julien-f) [inline src/index.js]: Use `xo.defineProperty('executor', executor)` instead.

And I believe the name jobExecutor would be clearer.

File: src/index.js [hunk 2]
```diff
@@ -442,9 +442,18 @@ const setUpApi = (webSocketServer, xo, verboseLogsOnErrors) => {
...
+  return executor
+}
+
 const setUpScheduler = xo => {
-  const jobExecutor = new JobExecutor(xo)
-  const scheduler = new Scheduler(xo, {executor: jobExecutor})
+  if (!xo.executor) {
+    setUpJobExecutor(xo)
+  }
+  const scheduler = new Scheduler(xo, {executor: xo.executor})
```
Comment (julien-f) [inline src/index.js]: Maybe it is no longer necessary to pass the job executor as it is available from `xo`.

File: src/xo.js [hunk 3]
```diff
@@ -647,6 +647,28 @@ export default class Xo extends EventEmitter {
     return await this._jobs.remove(id)
   }
 
+  async runJobSequence (idSequence) {
+    const notFound = []
+    forEach(idSequence, async id => {
```
Comment (julien-f) [inline src/xo.js]: Jobs are still run in parallel: `forEach()` does not wait for async functions to terminate before iterating the next items.

Comment (marsaud) [inline src/xo.js]: A while (id = idSequence.pop()) like ?...

Comment (julien-f) [inline src/xo.js]: Yep, or a `for (; ;)` or a `for ( of )`, just use a native syntax.

File: src/xo.js [hunk 4]
```diff
@@ -647,6 +647,28 @@ export default class Xo extends EventEmitter {
     return await this._jobs.remove(id)
   }
 
+  async runJobSequence (idSequence) {
+    const notFound = []
+    for (let id of idSequence) {
```
Comment (julien-f) [inline src/xo.js]: You can use `const` with `for ( of )`.

File: src/index.js (hunk 1 above)
Comment (marsaud) [inline src/index.js]: Can you explain me quickly the benefit to use defineProperty in this case ? (no emergency)

Comment (julien-f) [inline src/index.js]: There is no major benefits.

I am just trying to provide a standard interface to extend an `xo` instance from *the outside*.

Minor benefits:
- the property cannot be overwritten
- if the property is a function, it does not have `xo` for context and cannot pollute accidentally

File: src/index.js (hunk 2 above)
Comment (marsaud) [inline src/index.js]: I don't think the Scheduler class can take that for sure... Scheduler uses xo and and the executor as 2 different components. The executor is only available in xo by today's architecture choices...

Comment (julien-f) [inline src/index.js]: As you wish.

**Pull request merged**